internal/nprintf/nprintf.go

```diff
@@ -0,0 +1,23 @@
+package nprintf
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+// NPrintf is a helper function to replace placeholders in a string with values from a map
```

Comment:
```suggestion
// NPrintf is a helper function to replace placeholders in a string with values from a map
//
// Deprecated: nprintf.NPrintf exists to support migrating resources and data source out
// of the equinix package and into separate packages under `internal/resources`.  For new
// code, prefer to use fmt.Sprintf which enables more explicit and flexible formatting at the
// potential cost of higher repetition.
```
Nit: I lean towards deprecating this function entirely, in which case we should discouraged new usage, but that could be a separate follow-up discussion.